packages/admin/admin/src/common/buttons/clearinput/ClearInputButton.tsx

```diff
@@ -1,47 +1,55 @@
 import { Clear } from "@comet/admin-icons";
-import { ButtonBase, ButtonBaseClassKey, ButtonBaseProps, ComponentsOverrides, inputAdornmentClasses, Theme } from "@mui/material";
-import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import { ButtonBase, ButtonBaseProps, ComponentsOverrides } from "@mui/material";
+import { css, styled, Theme, useThemeProps } from "@mui/material/styles";
 import * as React from "react";
 
-export type ClearInputButtonClassKey = ButtonBaseClassKey;
+export type ClearInputButtonClassKey = "root" | "focusVisible";
+
+const Root = styled(ButtonBase, {
+    name: "CometAdminClearInputButton",
+    slot: "root",
+    overridesResolver(_, styles) {
+        return [styles.root];
+    },
+})(
+    ({ theme }) => css`
+        height: 100%;
+        width: 40px;
+        color: ${theme.palette.action.active};
+
+        &:last-child {
+            margin-right: ${theme.spacing(-2)};
+        }
+
+        &:first-child {
+            margin-left: ${theme.spacing(-2)};
+        }
```

Comment:
`:last-child` and `:first-child` is not correct here. 
`margin-right` should only be set when used inside the last `<InputAdornment position="end" />`, that's what `inputAdornmentClasses.positionEnd` was used for.